redpanda/cloud/ratelimiter.go

```diff
@@ -0,0 +1,94 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	limitPeriod = 60.0
+	burstPeriod = 10.0
+)
+
+type rateLimiter struct {
+	limiter *rate.Limiter
+}
+
+func newRateLimiter(limit int) *rateLimiter {
+	return &rateLimiter{
+		limiter: rate.NewLimiter(rate.Limit(limit/limitPeriod), limit/burstPeriod),
+	}
+}
+
+func parseRateLimit(header string) (limit, remaining int, reset time.Duration, err error) {
+	for _, part := range strings.Split(header, ",") {
+		kv := strings.SplitN(strings.TrimSpace(part), "=", 2)
+		if len(kv) != 2 {
+			continue
+		}
+		switch kv[0] {
```

Comment:
```suggestion
                keyword, val := kv[0], kv[1]
		switch keyword {
```

And then use the value in the error messages below